components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementServiceHolder.java

```diff
@@ -34,174 +39,221 @@
-    private static OAuthAdminServiceImpl oauthAdminService;
-    private static STSAdminServiceInterface stsAdminService;
-    private static SAMLSSOConfigServiceImpl samlssoConfigService;
-    private static OAuthServerConfiguration oAuthServerConfiguration;
-    private static TemplateManager templateManager;
-    private static CORSManagementService corsManagementService;
-    private static RealmService realmService;
-    private static APIResourceManager apiResourceManager;
-    private static AuthorizedAPIManagementService authorizedAPIManagementService;
-    private static OrgApplicationManager orgApplicationManager;
-
-    private static class LoginFlowAIManagerServiceHolder {
+    private ApplicationManagementServiceHolder() {
 
-        static final LoginFlowAIManager SERVICE =
-                (LoginFlowAIManager) PrivilegedCarbonContext.getThreadLocalCarbonContext()
-                        .getOSGiService(LoginFlowAIManager.class, null);
     }
 
-    public static ApplicationManagementService getApplicationManagementService() {
+    private static class ApplicationServiceHolder {
 
-        return applicationManagementService;
+        static final ApplicationManagementService SERVICE = (ApplicationManagementService) PrivilegedCarbonContext
+                .getThreadLocalCarbonContext().getOSGiService(ApplicationManagementService.class, null);
     }
 
-    public static void setApplicationManagementService(ApplicationManagementService applicationManagementService) {
+    private static class OAuthAdminServiceImplHolder {
 
-        ApplicationManagementServiceHolder.applicationManagementService = applicationManagementService;
+        static final OAuthAdminServiceImpl SERVICE = (OAuthAdminServiceImpl) PrivilegedCarbonContext
+                .getThreadLocalCarbonContext().getOSGiService(OAuthAdminServiceImpl.class, null);
     }
 
-    public static OAuthAdminServiceImpl getOAuthAdminService() {
+    private static class STSAdminServiceInterfaceHolder {
+
+        private static final Log log = LogFactory.getLog(STSAdminServiceInterfaceHolder.class);
+        static final STSAdminServiceInterface SERVICE = initializeService();
+
+        /**
+         * Initializes the STSAdminServiceInterface from the OSGi context.
+         *
+         * @return STSAdminServiceInterface or null if the service is unavailable.
+         */
+        private static STSAdminServiceInterface initializeService() {
+
+            try {
+                STSAdminServiceInterface service = (STSAdminServiceInterface) PrivilegedCarbonContext
+                        .getThreadLocalCarbonContext().getOSGiService(STSAdminServiceInterface.class, null);
+
+                if (service == null) {
+                    // Log a warning if the service is not available, but don't throw an exception
+                    log.warn("STSAdminServiceInterface is not available from OSGi context. This may occur if " +
```

Comment:
Properly handledthe service being null

Reply:
This is implemented following the previous implementation of the STSAdminOSGiServiceFactory